benchopt/tests/test_config.py

```diff
@@ -13,8 +14,15 @@
 @contextmanager
 def temp_config_file(permission='600'):
     permission = int(f"100{permission}", base=8)
-    config_file = Path() / 'test_config_file.yml'
-    config_file.touch(mode=permission, exist_ok=False)
+    config_file = Path(Path() / 'test_config_file.yml')
+    if sys.platform != 'win32':
+        config_file.touch(mode=permission, exist_ok=False)
+    else:
+        config_file.touch(exist_ok=False)
+        if permission == 0o600:
```

Comment:
if permission is something else, we put 666 by default, is it what we want?

Reply:
Yes I believe changing the `temp_file_config()` function to default on `permission=666` would make the test pass, but I don't know if it's fine to create config files that are writable by others.

The error message for this test is the following : 

```
UserWarning: BenchOpt config file test_config_file.yml is with mode 666.
As it stores sensitive information such as the github token,
it is advised to use mode 600 (user rw only).
```

So I believe `permission=600` is what's recommended.

The test `test_config_file_permission_no_warning` calls `temp_config_file()` with no parameters, so it should pass the default `permission=600`. Yet, it doesn't. One explanation in the forum I linked is that "Windows only support `stat.S_IWRITE` and `stat.S_IREAD`. The other bits are ignored.". So on Windows, even if we set the file to permission 600, it will still set itself to 666 internally.

Either this test should be skipped, or its code should be changed so that the default permission is 666.

Reply:
In the config file, we are stocking the GitHub token of the user, which should not be readable by other users.
This is why we are using a file that is only accessible to the user with permission `600` on linux.
I would say it is okay to skip the tests on windows if there is no way to make the file only readable by the user.

But we should put a warning when reading the config file if the `github_token` is set on windows as this is not safe.